core/consensus/babe/src/lib.rs

```diff
@@ -463,8 +463,9 @@ fn find_next_epoch_digest<B: BlockT>(header: &B::Header) -> Result<Option<Epoch>
 /// unsigned.  This is required for security and must not be changed.
 ///
 /// This digest item will always return `Some` when used with `as_babe_pre_digest`.
-// FIXME #1018 needs misbehavior types
-fn check_header<B: BlockT + Sized, C: AuxStore>(
+// FIXME #1018 needs misbehavior types. The `transaction_pool` parameter will be 
+// used to submit such misbehavior reports.
+fn check_header<B: BlockT + Sized, C: AuxStore, T>(
```

Comment:
Shouldn't we restrict the `T` type here, add some bounds to make sure it is some transaction pool type? Maybe actually receive a `Option<&T>` here since it seems like the transaction pool will be optional, and we just won't send reports if it is not available.

Reply:
Yes, we should restrict `T` with a trait that still needs to be defined (IMO this also depends on  #3063). So I prefer to leave it open for now. 

I will change it to `Option<&T>`.

Reply:
https://github.com/paritytech/substrate/blob/master/core/network/src/service.rs#L65

Reply:
We also need to create the tx here, right?

I'm not sure it works, will try to check.